tests/v2/validation/deleting/delete_cluster_test.go

```diff
@@ -0,0 +1,93 @@
+package deleting
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters/kubernetesversions"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioning"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioninginput"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	password "github.com/rancher/rancher/tests/framework/extensions/users/passwordgenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ClusterDeleteTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	standardUserClient *rancher.Client
+	clustersConfig     *provisioninginput.Config
+}
+
+func (c *ClusterDeleteTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *ClusterDeleteTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+
+	c.clustersConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, c.clustersConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(c.T(), err)
+
+	c.client = client
+
+	c.clustersConfig.K3SKubernetesVersions, err = kubernetesversions.Default(c.client, clusters.K3SClusterType.String(), c.clustersConfig.K3SKubernetesVersions)
+	require.NoError(c.T(), err)
+
+	c.clustersConfig.RKE1KubernetesVersions, err = kubernetesversions.Default(c.client, clusters.RKE1ClusterType.String(), c.clustersConfig.RKE1KubernetesVersions)
+	require.NoError(c.T(), err)
+
+	c.clustersConfig.RKE2KubernetesVersions, err = kubernetesversions.Default(c.client, clusters.RKE2ClusterType.String(), c.clustersConfig.RKE2KubernetesVersions)
+	require.NoError(c.T(), err)
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(c.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(c.T(), err)
+
+	c.standardUserClient = standardUserClient
+}
+
+func (c *ClusterDeleteTestSuite) TestDeletingRKE2K3SCluster() {
```

Comment:
Since we're running tests in recurring runs by tag, this wouldn't work if we ran the whole test suite. We can't design test suites where we can only run on test case at a time. This needs to be separated.

Reply:
Got it, I will split this into separate RKE2/K3S functions